README.md

```diff
@@ -199,6 +200,86 @@ metrics = require('metrics')
 
+#### Average
+  Can be used only as HTTP statistics collector (described below) and cannot be built explicitly.
+
+#### `average_obj:collect()`
+  Returns average value of observations for the observing period (time from previous collect call to now).
+  Also returns observation count for the same period.
+
+### Collecting HTTP requests latency statistics
+
+`metrics` also provides a middleware for monitoring HTTP (set by [http](https://github.com/tarantool/http) module) latency statistics.
+
+```
+-- importing submodule
+http_middleware = metrics.http_middleware
+```
+
+#### `http_middleware.configure_default_collector(type_name, name, help)`
+  Builds collector for middleware and sets it as default.
+  * `type_name` Collector type (string): "histogram" or "average". Default is "histogram".
+  * `name` Collector name (string). Default is "http_server_request_latency".
+  * `help` (optional) Help description (string). Default is "HTTP Server Request Latency".
+
+  If collector already exists, uses it. Otherwise, registers new collector.
+
+#### `http_middleware.build_default_collector(type_name, name, help)`
+  Builds collector for middleware.
+  * `type_name` Collector type (string): "histogram" or "average". Default is "histogram".
+  * `name` Collector name (string). Default is "http_server_request_latency".
+  * `help` (optional) Help description (string). Default is "HTTP Server Request Latency".
+
+  If collector already exists, returns it. Otherwise, registers new collector.
+
+#### `http_middleware.set_default_collector(collector)`
+  Set default collector.
+  * `collector` Middleware collector object.
+
+#### `http_middleware.get_default_collector()`
+  Returns default collector.
+  If default collector hasn't been set yet, builds it (with default build parameters) and sets it as default.
+
+#### `http_middleware.v1(handler, collector)`
+  Latency measure wrap-up for HTTP ver. 1.x.x handler. Returns wrapped handler.
+  * `handler` Handler function
+  * `collector` Middleware collector object. If not set, uses default collector (like in `http_middleware.get_default_collector()`).
+
+  Usage:
+  ```
+  httpd:route(route, http_middleware.v1(request_handler, collector))
+  ```
+
+  For more detailed example see [example/HTTP/latency_v1.lua](./example/HTTP/latency_v1.lua).
+
+#### `http_middleware.v2(collector)`
+  Returns latency measure middleware for HTTP ver. 2.x.x.
+  * `collector` Middleware collector object. If not set, uses default collector (like in `http_middleware.get_default_collector()`).
+
+  Usage:
+  ```
+  router = require('http.router').new()
+
+  router:route(route, request_handler)
+  router:use(http_middleware.v2(collector), {name = 'http_instrumentation'}) -- Second argument is optional, see HTTP docs
+  ```
+
+  For more detailed example see [example/HTTP/latency_v2.lua](./example/HTTP/latency_v2.lua).
+
+#### `http_middleware.observe(collector, route, ...)`
```

Comment:
I think this one can be omitted.

Reply:
Remove `http_middleware.observe(...)` description